test/integration/apiserver/admissionwebhook/validatingadmissionpolicy_test.go

```diff
@@ -0,0 +1,733 @@
+			{
+				RuleWithOperations: admissionregistrationv1.RuleWithOperations{
+					Operations: []admissionregistrationv1.OperationType{admissionregistrationv1.OperationAll},
+					Rule:       admissionregistrationv1.Rule{APIGroups: []string{"*"}, APIVersions: []string{"*"}, Resources: []string{"*/*"}},
+				},
+			},
+		},
+		MatchPolicy: &exact,
+	}
+
+	convertedPolicyTemplate.Spec.MatchConstraints = &admissionregistrationv1alpha1.MatchResources{
+		ResourceRules: convertedRules,
+		MatchPolicy:   &equivalent,
+	}
+
+	exactPolicy, err := client.AdmissionregistrationV1alpha1().ValidatingAdmissionPolicies().Create(context.TODO(), exactPolicyTemplate, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	convertPolicy, err := client.AdmissionregistrationV1alpha1().ValidatingAdmissionPolicies().Create(context.TODO(), convertedPolicyTemplate, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	// Create a param that holds the options for this
+	configuration, err := client.CoreV1().ConfigMaps("default").Create(context.TODO(), &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-policy-v1alpha1-param-",
+			Namespace:    "default",
+			Annotations: map[string]string{
+				"skipMatch": "yes",
+			},
+		},
+		Data: map[string]string{
+			"version": "v1alpha1",
+			"phase":   validation,
+			"convert": "false",
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	configurationConvert, err := client.CoreV1().ConfigMaps("default").Create(context.TODO(), &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-policy-v1alpha1-convert-param-",
+			Namespace:    "default",
+			Annotations: map[string]string{
+				"skipMatch": "yes",
+			},
+		},
+		Data: map[string]string{
+			"version": "v1alpha1",
+			"phase":   validation,
+			"convert": "true",
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	_, err = client.AdmissionregistrationV1alpha1().ValidatingAdmissionPolicyBindings().Create(context.TODO(), &admissionregistrationv1alpha1.ValidatingAdmissionPolicyBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-policy-v1alpha1-binding-",
+		},
+		Spec: admissionregistrationv1alpha1.ValidatingAdmissionPolicyBindingSpec{
+			PolicyName:        exactPolicy.GetName(),
+			ValidationActions: []admissionregistrationv1alpha1.ValidationAction{admissionregistrationv1alpha1.Warn},
+			ParamRef: &admissionregistrationv1alpha1.ParamRef{
+				Name:                    configuration.GetName(),
+				Namespace:               configuration.GetNamespace(),
+				ParameterNotFoundAction: &denyAction,
+			},
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+	_, err = client.AdmissionregistrationV1alpha1().ValidatingAdmissionPolicyBindings().Create(context.TODO(), &admissionregistrationv1alpha1.ValidatingAdmissionPolicyBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-policy-v1alpha1-convert-binding-",
```

Comment:
Why have to use this format and generatedName? Could we use `name` directly? Same question applied to other places.

Reply:
More of a historical detail at this point I suppose. I can change to use direct names now. Not sure what you refer to by this "format"? Is there something I should change the name to?

Reply:
The format I referred is regarding with the requirement we ask for name to satisfy `RFC 1123 subdomain`. :)